### DISCUSSION TITLE
How to use linq select in template?

### DISCUSSION THREAD

**Discussion by biplan66** [adoconnection/RazorEngineCore#76] [Q&A] [answered]
I have an array from an external assembly that I want to output.
So, I want to print List with indexes.
```
@foreach (var item in Model.Collection.Select((value, int i) => return new { i, value }) 
{
<div>@item.value</div>
<div>@item.i</div>
}
```
I got an error: `Unable to compile template: om00yyhz.oz1(48,131): error CS1977: Cannot use a lambda expression as an argument to a dynamically dispatched operation`
How can I fix it?

**Comment by adoconnection** [comment, marked as answer]
https://stackoverflow.com/questions/27169521/cannot-use-a-lambda-expression-as-an-argument-to-a-dynamically-dispatched-operat

**Reply by biplan66** [reply]
Thanks!)

**Comment by biplan66** [comment]
What am I doing wrong?
I got an error: `System.Collections.Generic.List does not contain a definition for 'Select'`
And I use builder and addAssemblyReference `builder.AddAssemblyReference(typeof(Enumerable).Assembly);`

**Reply by biplan66** [reply]
So, I use for and foreach to solve my problem)) 
How can I connect an assembly to use the Select method?

**Reply by adoconnection** [reply]
Yep, please refer to this unit test code: https://github.com/adoconnection/RazorEngineCore/blob/2b1f2901e609118e140996ad6b850dfab7665e1e/RazorEngineCore.Tests/TestCompileAndRun.cs#L540
